samples/c/create_systemd_scope.c

```diff
@@ -0,0 +1,614 @@
+ */
+
+/*
+ * To compile and link this program:
+ * 	(From the root of the libcgroup source code directory)
+ * 	$ ./bootstrap
+ * 	$ ./configure --sysconfdir=/etc --localstatedir=/var \
+ * 	  --enable-opaque-hierarchy="name=systemd" --enable-systemd \
+ * 	  --enable-python --enable-samples
+ * 	$ make
+ *
+ * Add the libcgroup idle thread to your PATH.  (Some distros restrict the
+ * modification of the $PATH environment variable when invoking sudo, so you
+ * will need to manually copy the executable to your path.)
+ * 	$ sudo cp src/libcgroup_systemd_idle_thread /a/path/in/your/sudo/path
+ *
+ * To run this program:
+ *      $ # Note that there are more options.  Run `create_systemd_scope -h` for more info
+ * 	$ sudo LD_LIBRARY_PATH=src/.libs ./samples/c/create_systemd_scope \
+ * 	  --slice <yourslicename> --scope <yourscopename>
+ */
+
+/*
+ */
+
+#include <libcgroup.h>
+#include <sys/wait.h>
+#include <getopt.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+
+#define TMP_CGNAME "tmp"
+#define HIGH_CGNAME "high-priority"
+#define MED_CGNAME "medium-priority"
+#define LOW_CGNAME "low-priority"
+
+struct example_opts {
+	char slice[FILENAME_MAX];
+	char scope[FILENAME_MAX];
+
+	struct cgroup_systemd_scope_opts systemd_opts;
+
+	bool debug;
+};
+
+static const struct option long_opts[] = {
+	{"delegated",		no_argument,	   NULL, 'd'},
+	{"help",		no_argument,	   NULL, 'h'},
+	{"pid",			required_argument, NULL, 'p'},
+	{"scope",		required_argument, NULL, 's'},
+	{"slice",		required_argument, NULL, 't'},
+	{"verbose",		no_argument,       NULL, 'v'},
+	{NULL, 0, NULL, 0}
+};
+
+static void usage(const char * const program_name)
+{
+	printf("Usage: %s -s <scope> -t <slice> [-p <pid>]\n", program_name);
+	printf("Create a systemd scope using the libcgroup C APIs\n");
+	printf("  -d, --delegated	Instruct systemd to create a delegated scope\n");
+	printf("  -p, --pid		PID to be placed in the scope, if not provided,\n");
+	printf("			libcgroup will place a default idle PID in the scope\n");
+	printf("  -s, --scope		Scope name, must end in .scope\n");
+	printf("  -t, --slice		Slice name, must end in .slice\n");
+	printf("  -v, --verbose		Enable libcgroup debug messages\n");
+}
+
+static int parse_opts(int argc, char *argv[], struct example_opts * const opts)
+{
+	int ret = 0;
+	int c;
+
+	while ((c = getopt_long(argc, argv, "hs:t:p:dv", long_opts, NULL)) > 0) {
+		switch (c) {
+		case 'd':
+			opts->systemd_opts.delegated = true;
+			break;
+		case 'h':
+			usage(argv[0]);
+			exit(0);
+		case 'p':
+			opts->systemd_opts.pid = atoi(optarg);
+			if (opts->systemd_opts.pid <= 1) {
+				usage(argv[0]);
+				exit(1);
+			}
+			break;
+		case 's':
+			strncpy(opts->scope, optarg, FILENAME_MAX - 1);
+			break;
+		case 't':
+			strncpy(opts->slice, optarg, FILENAME_MAX - 1);
+			break;
+		case 'v':
+			opts->debug = true;
+			break;
+		default:
+			usage(argv[0]);
+			exit(1);
+			break;
+		}
+	}
+
+	return ret;
```

Comment:
Minor nitpick `ret` is unused, should it be `return 0`?

Reply:
Good call.  I'll change it.